luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/factories/KeyValueStoreDimensionFactory.java

```diff
@@ -0,0 +1,89 @@
+package com.yahoo.bard.webservice.config.luthier.factories;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.yahoo.bard.webservice.config.luthier.Factory;
+import com.yahoo.bard.webservice.config.luthier.LuthierIndustrialPark;
+import com.yahoo.bard.webservice.data.config.LuthierDimensionField;
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionField;
+import com.yahoo.bard.webservice.data.dimension.KeyValueStore;
+import com.yahoo.bard.webservice.data.dimension.SearchProvider;
+import com.yahoo.bard.webservice.data.dimension.impl.KeyValueStoreDimension;
+import com.yahoo.bard.webservice.util.EnumUtils;
+
+import java.util.LinkedHashSet;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class KeyValueStoreDimensionFactory implements Factory<Dimension> {
+
+
+    /**
+     * @param fieldsNode the JsonNode object that points to the content of "fields" key
+     *                   or "defaultFields" key
```

Comment:
We should not be using the same function for `defaultFields` as `fields`. `defaultFields` is a list of
strings, *not* a list of tables containing field definitions. Parsing the `defaultFields` should be as
simple as iterating over `defaultFields`, and looking up each one in a `Map<String, DimensionField>` of the
actual dimension fields. Note that this means you'll probably want to parse the `fields` into a `LinkedHashMap`,
use it to hydrate thee `defaultFields` and *then* convert its `values()` into a `LinkedHashSet` for passing
to the Dimension.

Reply:
My bad. I did not see this comment when addressing the one two above. 
Is the up-stream going to accept some kind of LuthierFactoryException or should I just throw a runtime error when a string in defaultFields is not a key we built before?

Reply:
Throw a LuthierFactoryException with a useful message, like "Attempted to use unknown field '%s' as a default field." or something.